### DISCUSSION TITLE
How to Retreive the Distance Matrix

### DISCUSSION THREAD

**Discussion by iamdavie** [TDAmeritrade/stumpy#405] [Q&A] [answered]
As posted in #349 , Is there a way to just retrieve the distance matrix, so that one can design a way to retrieve the top-K matches for EVERY sub sequence within their time series. ??

**Comment by seanlaw** [comment, marked as answer]
@iamdavie No, unfortunately, it is not possible to return the distance matrix with STUMPY as this is beyond goals/scope of computing matrix profiles according to the original authors of the matrix profile work. For a time series with length `n`, storing the distance matrix requires `O(n*n)` space. So, even a short time series where `n = 100,000` this would require roughly 80 GB of memory. The point of the matrix profile data structure is to be `O(n)` in space complexity.

**Reply by iamdavie** [reply]
Thank you for the quick reply on such a short notice, I was so focused on the time complexity that i ignored the memory constraints!!

**Reply by seanlaw** [reply]
It was a valid question. I've reached out to the original authors of the matrix profile work to see if they have some ideas.

**Reply by iamdavie** [reply]
Guess, We just have to wait and see, but thumbs up!!, Excellent work on STUMPY!!
